jetty-core/jetty-http-spi/src/main/java/org/eclipse/jetty/http/spi/JettyHttpExchangeDelegate.java

```diff
@@ -68,9 +69,12 @@ public Headers getRequestHeaders()
 
         for (HttpField field : _request.getHeaders())
         {
-            if (field.getValue() == null)
+            String rawValue = field.getValue();
+            if (rawValue == null)
                 continue;
-            for (String value : field.getValues())
+            // Using raw QuotedCSV here to preserve quotes (which HttpField.getValues() doesn't do)
+            QuotedCSV quotedCSV = new QuotedCSV(true, rawValue);
+            for (String value : quotedCSV.getValues())
```

Comment:
Not all headers can be parsed with `QuotedCSV`, for example `User-Agent: ... (KHTML, like Gecko)...`.
`Expires` too has a comma, but it's only a response field.
Not sure if there are request fields that have dates in them, maybe `Date`, `If-Modified-Since`, etc.

Reply:
I've updated the PR.
On review of RFC9110, the following list of headers do not have value lists.

https://github.com/eclipse/jetty.project/blob/b64c2b431ea01f26fc17d7bab68cfa7c4fd0dd4d/jetty-core/jetty-http-spi/src/main/java/org/eclipse/jetty/http/spi/JettyHttpExchangeDelegate.java#L80-L97